tests/Feature/Platform/Action/WriteGameSortTitleFromGameTitleActionTest.php

```diff
@@ -0,0 +1,147 @@
+        );
+        $game = $game->fresh();
+
+        // Assert
+        $this->assertEquals('final fantasy 04', $game->sort_title);
+    }
+
+    public function testItCorrectlyHandlesGameTitlesWithTildes(): void
+    {
+        // Arrange
+        $gameOne = Game::factory()->create(['Title' => '~Homebrew~ Classic Kong', 'sort_title' => null]);
+        $gameTwo = Game::factory()->create(['Title' => 'Puyo Puyo~n', 'sort_title' => null]);
+
+        // Act
+        (new WriteGameSortTitleFromGameTitleAction())->execute($gameOne, $gameOne->title);
+        (new WriteGameSortTitleFromGameTitleAction())->execute($gameTwo, $gameTwo->title);
+        $gameOne = $gameOne->fresh();
+        $gameTwo = $gameTwo->fresh();
+
+        // Assert
+        $this->assertEquals('~homebrew classic kong', $gameOne->sort_title);
+        $this->assertEquals('puyo puyo~n', $gameTwo->sort_title);
+    }
+
+    public function testItCorrectlyConvertsRomanNumerals(): void
+    {
+        // Arrange
+        $gameOne = Game::factory()->create(['Title' => 'Final Fantasy IV', 'sort_title' => null]);
+        $gameTwo = Game::factory()->create(['Title' => 'Final Fantasy X', 'sort_title' => null]);
+
+        // Act
+        (new WriteGameSortTitleFromGameTitleAction())->execute($gameOne, $gameOne->title);
+        (new WriteGameSortTitleFromGameTitleAction())->execute($gameTwo, $gameTwo->title);
+        $gameOne = $gameOne->fresh();
+        $gameTwo = $gameTwo->fresh();
+
+        // Assert
+        $this->assertEquals('final fantasy 04', $gameOne->sort_title);
+        $this->assertEquals('final fantasy 10', $gameTwo->sort_title);
+    }
+
+    public function testItAvoidsNonRomanStrings(): void
+    {
+        // Arrange
+        $game = Game::factory()->create(['Title' => 'GIVX', 'sort_title' => null]);
+
+        // Act
+        (new WriteGameSortTitleFromGameTitleAction())->execute($game, $game->title);
+        $game = $game->fresh();
+
+        // Assert
+        $this->assertEquals('givx', $game->sort_title);
+    }
+
+    public function testItNormalizesArticleFragments(): void
+    {
+        // Arrange
+        $gameOne = Game::factory()->create(['Title' => 'Legend of Zelda, The: A Link to the Past', 'sort_title' => null]);
+        $gameTwo = Game::factory()->create(['Title' => 'American Tale, An', 'sort_title' => null]);
+        $gameThree = Game::factory()->create(['Title' => 'Grand Day Out, A', 'sort_title' => null]);
+
+        // Act
+        (new WriteGameSortTitleFromGameTitleAction())->execute($gameOne, $gameOne->title);
+        (new WriteGameSortTitleFromGameTitleAction())->execute($gameTwo, $gameTwo->title);
+        (new WriteGameSortTitleFromGameTitleAction())->execute($gameThree, $gameThree->title);
+        $gameOne = $gameOne->fresh();
+        $gameTwo = $gameTwo->fresh();
+        $gameThree = $gameThree->fresh();
+
+        // Assert
+        $this->assertEquals('legend of zelda: a link to the past', $gameOne->sort_title);
```

Comment:
Sorry, for some reason this feedback isn't clicking in my brain. Can you restate this feedback?

Reply:
Ah, I understand now and this is intentional, though we can change it if you feel stripping the article in this way is desirable. There are even a few test cases to verify it's _not_ stripped at the moment:

```php
            'An Unexpected Journey' => ['An Unexpected Journey', 'an unexpected journey'],
            'The Great Escape' => ['The Great Escape', 'the great escape'],
            'The Matrix Reloaded' => ['The Matrix Reloaded', 'the matrix reloaded'],
```

I was thinking it may be too difficult to tell algorithmically that an article shouldn't belong in the sort title if it's not separated by the comma. "The Matrix Reloaded" is a good example - "The Matrix" is a literal title for the work, and "The" is doing more work than as an article.

Reply:
"The Matrix" is already sorted under 'M': https://retroachievements.org/game/20823

And per https://github.com/RetroAchievements/RAWeb/discussions/1062:

> Ignore definite and indefinite articles **at the start of titles** so we don't have to rename them to force the correct sorting order.

Reply:
While the `The` is technically part of the title, most libraries ignore it because it would make their "T" section so much larger and people would still have to subsort on the next word to find what they're looking for. 

There are currently 454/9095 games with achievements that have a `, The` in their title. If we include the article in the sort, there would be 9 pages of `The` games. 

Additionally, [the brain often ignores articles](https://psychology.stackexchange.com/questions/13946/why-does-the-brain-skip-over-repeated-the-words-in-sentences), so the user might not even remember that the "The" is part of the title and look for it under the next word anyway.

And then you have some series that make flippant use of the article. There's a spinoff of "The King of Fighters '97" called "[King of Fighters R-1](https://github.com/RetroAchievements/RALibretro/issues/416)". Shouldn't they be sorted near each other?

`A` and `An` make this even harder because `An` only applies to vowels. `An Apple` should sort before `A Banana`, but wouldn't if we don't ignore the article.

Reply:
I guess the question you have to ask yourself is: If you walked into a rental store to borrow "The Matrix", would you look under "T" or "M"? I'd definitely look under "M".

Reply:
Resolved in latest.